src/components/Search/Result.spec.tsx

```diff
@@ -0,0 +1,25 @@
+import 'react-native'
+import React from 'react'
+import { render, fireEvent } from '@testing-library/react-native'
+
+import SearchResult from './Result'
+
+describe( '<SearchResult />', () => {
+  it( 'should handle an onPress event', () => {
+    const onEventMock = jest.fn()
+
+    const { getByTestId } = render(
+      <SearchResult
+        line="line"
+        source="source"
+        page="page"
+        translation="translation"
+        testID="search-result-test"
+        onPress={onEventMock}
+      />,
+    )
+
+    fireEvent.press( getByTestId( 'search-result-test' ) )
+    expect( onEventMock ).toHaveBeenCalled()
```

Comment:
```suggestion
    expect( onEventMock ).toHaveBeenCalledWith( { id: 'xas' } )
```

Or similar, but test the way that your component is intended to be used at a **reasonable** level of granularity.

Reply:
Can you please explain this little more? In this case the `onPress` event is just used by higher up component in this case the `Results` component is where we use this handler to update the state. This (`Result`) is just a stateless component

Reply:
I just mean, test it with some reasonable expectation of what your average callback might contain, data-wise. It serves as wonderful documentation and provides more robustness to your test

Reply:
Callback mutate state inside this component. I don’t think any data being passed to callback in this component. I don’t have good terminology to explain this. It is just exposing the onPress callback for the user to call and do tasks, there is nothing that needs to be mutated inside the component

Reply:
Oh, I understand. You're saying that onPress doesn't supply any data.

Reply:
Yes 👍🏼